app/lib/handlers/intent_request_handlers/transaction_intent_request_handler.rb

```diff
@@ -0,0 +1,122 @@
+module Handlers
+  module IntentRequestHandlers
+
+    class TransactionIntentRequestHandler < BaseRequestHandler
+
+      def process
+        send(get_intent_specific_method(intent['name']))
+      end
+
+      private
+
+      def items
+        @items ||= get_items_from_slots
+      end
+
+      def slots
+        @slots ||= intent['slots'].to_h
+      end
+
+      def get_intent_specific_method(intent_name)
+        {
+          'Add'        => :process_add_items,
+          'Remove'     => :process_remove_items,
+          'Clear'      => :process_clear_items
+        }[intent_name]
+      end
+
+      def process_add_items
+        Helpers::HandlerHelper.create_message_response(
+          message: "#{prepare_transaction_items_for_message(added_items)} were added",
+          card: {
+            type: 'Simple',
+            title: 'Added Food Items',
+            content: Helpers::HandlerHelper.prepare_items_for_card_without_date(added_items)
+          },
+          reprompt: 'Is there anything else I can help you with?',
+          end_session: false
+        )
+      end
+
+      def process_remove_items
+        Helpers::HandlerHelper.create_message_response(
+          message: "#{prepare_transaction_items_for_message(removed_items)} were removed",
+          card: {
+            type: 'Simple',
+            title: 'Removed Food Items',
+            content: prepare_removed_items_for_card(removed_items)
+          },
+          reprompt: 'Is there anything else I can help you with?',
+          end_session: false
+        )
+      end
+
+      def process_clear_items
+        Item.owned_by(user_id).delete_all
+        Helpers::HandlerHelper.create_message_response(
+          message: "All items have been removed",
+          reprompt: 'Is there anything else I can help you with?',
+          end_session: false
+        )
+      end
+
+      def added_items
+        @added_items ||= items.reject{ |k,v| v <= 0 }.map do |k,v|
+          details = Helpers::CategoryHelper.categorize k
+          item = Item.new(
+            name: k,
+            category_small: details[:small_category],
+            category_large: details[:large_category],
+            quantity: v,
+            warning_date: Time.now + details[:time_til_warning].days,
+            expiration_date: Time.now + details[:time_til_expiration].days,
+            user_id: user_id
+          )
+          if details[:time_til_warning] != 0
+            item[:warning_date] = Time.now + details[:time_til_warning].days
+          end
+          if details[:time_til_expiration] != 0
+            item[:expiration_date] = Time.now + details[:time_til_expiration].days
+          end
+          item.save
+          item
```

Comment:
I like to use `item.tap(&:save)` for things like this.  That syntax will call the method from the symbol on whatever's being tapped, then return the tapped thing.